Ch-#162414110 initial project setup update

tosinibrahim96 opened the pull request:
**What does this PR do?**
Set-up all required tools required for the project to begin

**Description of Task to be completed**
- Setup an empty react project
- Setup react router version 4 (React-router-dom)
- Setup Redux
- Setup SASS processing on the client
- Setup Semantic UI React
- Add normalize CSS
- Add ESlint with Airbnb Styles
- Update TravisCI to run tests
- Run codeclimate test coverage

**How should this be manually tested?**
* Clone this repository
* Open your terminal and checkout to `git checkout ch-initial-project-setup-#162414110`
* Run `npm install` to install all dependencies
* Run `npm start` to run the application on your local server

**What are the relevant pivotal tracker stories?**
https://www.pivotaltracker.com/story/show/162414110
https://www.pivotaltracker.com/story/show/162414114
 https://www.pivotaltracker.com/story/show/162414111
 https://www.pivotaltracker.com/story/show/162414117

Commit:
[ch-#162414110 create initial project setup]

- Setup an empty react project
- Setup react router version 4 (React-router-dom)
- Setup Redux
- Setup SASS processing on the client
- Setup Semantic UI React
- Add normalize CSS

Commit:
[ch-#162414110 create initial project setup]

- Setup an empty react project
- Setup react router version 4 (React-router-dom)
- Setup Redux
- Setup SASS processing on the client
- Setup Semantic UI React
- Add normalize CSS

Commit:
[ch-#162414110 create initial project setup]

- Setup an empty react project
- Setup react router version 4 (React-router-dom)
- Setup Redux
- Setup SASS processing on the client
- Setup Semantic UI React
- Add normalize CSS

Maxfurry commented on webpack/webpack.dev.js:
```diff
@@ -0,0 +1,12 @@
+  devtool: 'inline-source-map',
+  devServer: {
+    host: 'localhost',
+    port: 3000,
```
I would advise process.env.PORT || 3000 be used for the port so that it will not fail on Heroku.

codrex commented on src/pages/404/index.jsx:
```diff
@@ -0,0 +1,9 @@
+import React from 'react';
+
+export default function NotFound() {
```
Folder name should match module name

tosinibrahim96 replied:
I've changed that

codrex commented on src/pages/Home/test/index.test.js:
```diff
@@ -0,0 +1,10 @@
+import React from 'react';
```
This file should be home.test.js and not index.test.js

tosinibrahim96 replied:
Noted

codrex commented on src/reducers/index.js:
```diff
@@ -0,0 +1,5 @@
+import { combineReducers } from 'redux';
+
+export default combineReducers({
+  name: () => 'randomReducer'
```
`name: () => 'randomReducer'` ??

tosinibrahim96 replied:
The redux dev tools was not displaying anything so i had to test it by returning a value. I'll remove it now, since i'm sure it's working

codrex commented on utils/reduxdevtools.js:
```diff
@@ -0,0 +1,5 @@
+import { compose } from 'redux';
```
This should be in its own folder called `store`

tosinibrahim96 replied:
I'll move that

codrex commented on utils/enzyme.setup.js:
```diff
@@ -0,0 +1,6 @@
+import Enzyme from 'enzyme';
```
This should be in the root directory and not in the utiles folder

tosinibrahim96 replied:
Done

codrex commented on src/index.jsx:
```diff
@@ -0,0 +1,19 @@
+import composeEnhancers from '../utils/reduxdevtools';
+
+const store = createStore(reducers, composeEnhancers(applyMiddleware(thunk)));
+
```
```js
import reducers from './reducers';
import composeEnhancers from '../utils/reduxdevtools';

 const store = createStore(reducers, composeEnhancers(applyMiddleware(thunk)));
```
The above should be in the store folder.

codrex commented on .eslintignore:
```diff
@@ -0,0 +1,3 @@
+dist/* 
+webpack*.*
+test/*.test.js
```
why are you adding tests to this file?

tosinibrahim96 replied:
Mistake from my side, I'll change this

Commit:
[ch-#162414110 create initial project setup]

- Setup an empty react project
- Setup react router version 4 (React-router-dom)
- Setup Redux
- Setup SASS processing on the client
- Setup Semantic UI React
- Add normalize CSS
